extensions/procfs/ProcFsSerialization.h

```diff
@@ -0,0 +1,163 @@
+  serializer("Compressed Packets Transmitted", net_dev_data.getCompressedPacketsTransmitted());
+}
+
+void SerializeNetDevDataPerSec(const NetDevData& net_dev_data,
+                               const std::chrono::duration<double> duration,
+                               std::invocable<const char(&)[], const double> auto serializer) {
+  gsl_Expects(duration > 0ms);
+  serializer("Bytes Received/sec", net_dev_data.getBytesReceived()/duration.count());
+  serializer("Packets Received/sec", net_dev_data.getPacketsReceived()/duration.count());
+  serializer("Receive Errors/sec", net_dev_data.getReceiveErrors()/duration.count());
+  serializer("Receive Drop Errors/sec", net_dev_data.getReceiveDropErrors()/duration.count());
+  serializer("Receive Fifo Errors/sec", net_dev_data.getReceiveFifoErrors()/duration.count());
+  serializer("Receive Frame Errors/sec", net_dev_data.getReceiveFrameErrors()/duration.count());
+  serializer("Compressed Packets Received/sec", net_dev_data.getCompressedPacketsReceived()/duration.count());
+  serializer("Multicast Frames Received/sec", net_dev_data.getMulticastFramesReceived()/duration.count());
+
+  serializer("Bytes Transmitted/sec", net_dev_data.getBytesTransmitted()/duration.count());
+  serializer("Packets Transmitted/sec", net_dev_data.getPacketsTransmitted()/duration.count());
+  serializer("Transmit errors/sec", net_dev_data.getTransmitErrors()/duration.count());
+  serializer("Transmit drop errors/sec", net_dev_data.getTransmitDropErrors()/duration.count());
+  serializer("Transmit fifo errors/sec", net_dev_data.getTransmitFifoErrors()/duration.count());
+  serializer("Transmit collisions/sec", net_dev_data.getTransmitCollisions()/duration.count());
+  serializer("Transmit carrier losses/sec", net_dev_data.getTransmitCarrierLosses()/duration.count());
+  serializer("Compressed Packets Transmitted/sec", net_dev_data.getCompressedPacketsTransmitted()/duration.count());
+}
+
+void SerializeProcessStat(const ProcessStat& process_stat,
+                          std::invocable<const char(&)[], const uint64_t> auto uint64_t_serializer,
+                          std::invocable<const char(&)[], const std::string_view&> auto string_serializer) {
+  string_serializer("COMM", process_stat.getComm());
+  uint64_t_serializer("RES", process_stat.getMemory());
+  uint64_t_serializer("CPUTIME", process_stat.getCpuTime().count());
+}
+
+void SerializeNormalizedProcessStat(const ProcessStat& process_stat_start,
+                                    const ProcessStat& process_stat_end,
+                                    const std::chrono::duration<double> all_cpu_time,
+                                    std::invocable<const char(&)[], const uint64_t> auto uint64_t_serializer,
+                                    std::invocable<const char(&)[], const std::string_view&> auto string_serializer,
+                                    std::invocable<const char(&)[], const double> auto double_serializer) {
```

Comment:
I think a single serializer with an overloaded or templated call operator would be nicer.
You can use the widely used `overloaded` utility:
```

Reply:
Good idea, reworked it in https://github.com/apache/nifi-minifi-cpp/pull/1152/commits/84e464c7b9501293616bcae170c4a12a710f7a6d#diff-8a4db62b36fb7340e48e062700fa2ae5b9c3851563ef91e310a51b84ba0e44c4R30